### PR TITLE
Fixes #1336 - Add target=_blank to reported URL (only)

### LINKED ISSUE #1336: [Issue] target="_blank" for link in issue detail
Any thoughts?

### PR DESCRIPTION
**Pull request by miketaylr**
This PR fixes issue #1336

## Proposed PR background

This checks the parsed markdown tokens for the following pattern `**URL**:  `, followed by a `link_open` token, and if that matches, adds `target=_blank` to the anchor element.

---

### CODE REVIEW

**Comment by miketaylr** [comment]
You can verify by clicking on the 2 angry birds links at http://localhost:5000/issues/160 (with this branch running locally)

File: webcompat/static/js/lib/issues.js [hunk 1]
```diff
@@ -53,6 +53,20 @@ md.linkify.add("#", {
     match.url = "/issues/" + match.url.replace(/^#/, "");
   }
 });
+
+var checkIfReportedURLOpeningA = function(tokens) {
+  if (
+    tokens[0].markup === "**" &&
+    tokens[1].content === "URL" &&
```
Comment (miketaylr) [inline webcompat/static/js/lib/issues.js]: actually, let's make this fail faster by checking the first 2 tokens as an initial check.

**Comment by magsout** [comment]
@miketaylr it's on my todo list. Lot of working travel ;)

**Comment by miketaylr** [comment]
> @miketaylr it's on my todo list. Lot of working travel ;)

no rush!

File: webcompat/static/js/lib/issues.js [hunk 2]
```diff
@@ -61,7 +77,17 @@ var defaultLinkOpenRender =
...
 md.renderer.rules.link_open = function(tokens, idx, options, env, self) {
-  tokens[idx].attrPush(["rel", "nofollow"]);
+  var foundReportedURL = checkIfReportedURLOpeningA(tokens);
+  // only add target=_blank to a URL that was parsed from the following:
+  // **URL**: <url>
+  if (foundReportedURL) {
+    tokens[idx].attrPush(["rel", "nofollow"]);
+    tokens[idx].attrPush(["target", "_blank"]);
+  } else {
+    tokens[idx].attrPush(["rel", "nofollow"]);
```
Comment (magsout) [inline webcompat/static/js/lib/issues.js]: Adding this line once before or after the if?

File: webcompat/static/js/lib/issues.js [hunk 3]
```diff
@@ -53,6 +53,22 @@ md.linkify.add("#", {
     match.url = "/issues/" + match.url.replace(/^#/, "");
   }
 });
+
+var checkIfReportedURLOpeningA = function(tokens) {
+  // fast-check the first 2 tokens, returning false if we don't match
+  if (tokens[0].markup !== "**" && tokens[1].content !== "URL") {
```
Comment (magsout) [inline webcompat/static/js/lib/issues.js]: not sure to understand this test. We should test:
```
if (tokens[0].markup == "**" && tokens[1].content == "URL") 
```

We want to add an attr `target="_blank"` for the URl reported by the author, isn'it ?

File: webcompat/static/js/lib/issues.js [hunk 4]
```diff
@@ -61,7 +77,17 @@ var defaultLinkOpenRender =
...
+  // only add target=_blank to a URL that was parsed from the following:
+  // **URL**: <url>
+  if (foundReportedURL) {
+    tokens[idx].attrPush(["rel", "nofollow"]);
+    tokens[idx].attrPush(["target", "_blank"]);
+  } else {
+    tokens[idx].attrPush(["rel", "nofollow"]);
+  }
+
+  //tokens[].attrPush(["rel", "lolfollow"]);
```
Comment (magsout) [inline webcompat/static/js/lib/issues.js]: nit: is this comment meaningful?

Comment (miketaylr) [reply webcompat/static/js/lib/issues.js]: haha no.

File: webcompat/static/js/lib/issues.js (hunk 2 above)
Comment (miketaylr) [reply webcompat/static/js/lib/issues.js]: +1

**Comment by magsout** [comment]
r+ @miketaylr 👍

**Comment by magsout** [comment]
@miketaylr I have click on the button "update branch", because it was out of date

**Comment by miketaylr** [comment]
Thanks!

**Pull request merged**